documentation/modules/ROOT/pages/modules/cdc-mysql-connector/c_mysql-connector-events.adoc

```diff
@@ -351,32 +374,57 @@ CREATE TABLE customers (
       "thread": 7,
       "query": "UPDATE customers SET first_name='Anne Marie' WHERE id=1004"
     },
-    "op": "u", <2>
+    "op": "u",
     "ts_ms": 1465581029523
   }
 }
 ----
 
+Comparing this to the value in the _insert_ event, you can see a couple of differences in the `payload` section:
+
+* The `op` field value is now `u`, signifying that this row changed because of an update.
+
+* The `before` field now has the state of the row with the values before the database commit.
+
+* The `after` field now has the updated state of the row, and the `first_name` value is now `Anne Marie`.
+
+* The `source` field structure has the same fields as before, but the values are different (this event is from a different position in the binlog).
+
+* The `ts_ms` field shows the timestamp when {prodname} processed this event.
+
+There are a couple things you can learn just by looking at the `payload` section:
+
+* You can compare the `before` and `after` structures to determine what actually changed in this row because of the commit.
+
+* The `source` structure shows information about MySQL’s record of this change (providing traceability). It also has information you can use to compare to other events in this and other topics to know whether this event occurred before, after, or as part of the same MySQL commit as other events.
+
+[NOTE]
+====
+When the columns for a row’s primary or unique key are updated, the value of the row’s key is changed and {prodname} outputs three events: a _DELETE_ event and tombstone event with the old key for the row, followed by an _INSERT_ event with the new key for the row.
+====
 
 === Delete change event value
 
+The value of a _delete_ change event on the `customers` table has the exact same schema as _create_ and _update_ events. The payload is structured the same, but holds different values. Here is an example (formatted for readability):
+
+
 [source,json]
 ----
 {
   "schema": { ... },
   "payload": {
-    "before": { <3>
+    "before": {
       "id": 1004,
       "first_name": "Anne Marie",
       "last_name": "Kretchmar",
       "email": "annek@noanswer.org"
     },
-    "after": null, <4>
-    "source": { <5>
+    "after": null,
+    "source": {
       "version": "0.10.0.Beta4",
```

Comment:
Could you use the attribute (see the JIRA issue for the name) instead of the hard-coded version (here and in the other listings).